README.md

```diff
@@ -821,64 +830,67 @@ class C {
 }
 ```
 
-Would produce the following two metadata objects:
+You would be able to access the metadata for this class like so:
 
 ```js
-let staticMeta = {
-  public: {
-    constructor: { "my-meta": true, [MY_META]: true },
-    x: { "my-meta": true, [MY_META]: true },
-    y: { "my-meta": true, [MY_META]: true },
-  },
+C[Symbol.metadata][MY_META];
+// {
+//   constructor: 'metadata',
+
+//   public: {
+//     x: 'metadata',
+//     y: 'metadata',
+//   },
+
+//   private: ['metadata']
+// }
+
+C[Symbol.metadata][MY_META];
+// {
+//   public: {
+//     a: 'metadata',
+//     b: 'metadata',
+//   },
+
+//   private: ['metadata']
+// }
+```
 
-  private: {
-    "#z": { "my-meta": true, [MY_META]: true },
-  }
-}
+Subsequent definitions to the same key will result in an array of values instead of a single value. This allows metadata to be built up with multiple decorators without having to _read_ the metadata during decoration.
```

Comment:
For context, the motivation for this API choice is performance, allowing us to avoid allocating an array for each property which has metadata.

For engines/implementations, this should be accomplishable with a flag or some other method of tracking the values internally, if that's the concern.

For users, most usages of metadata that we have seen are specific metadata formats that are unique to a particular decorator/group of decorators. There isn't really a common case where you may have arbitrary data associated with the metadata for a given decorator. So, since the metadata is generally well formed, it should be possible to design formats where the difference is clear. For instance, using an object rather than an array if you expect to have multiple values on the same key.

Reply:
After further thought, and while writing the spec, I realized that we can actually solve this problem better with a `getMetadata` method that would allow users to access the existing metadata if it exists. This would only allow users to access metadata for the value that was being decorated, so it wouldn't allow users to query metadata in general. I've updated this PR with these changes.